fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -502,6 +502,10 @@ private Statistics computeOlapScan(OlapScan olapScan) {
                 }
                 checkIfUnknownStatsUsedAsKey(builder);
                 builder.setRowCount(selectedPartitionsRowCount);
+            } else {
```

Comment:
If the range is not splitted as even distribution, the following will have skew on estimation; if yes, then the filter estimation based on partition key will have similar impact on partition estimation; so seems the added logic seems have no improvement?